src/Auth/UserAuth.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace App\Auth;
+
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use App\Repository\UserRepository;
+use App\Entity\User;
+
+class UserAuth
+{
+    /** @var SessionInterface */
+    private $session;
+
+    /** @var UserRepository */
+    private $userRepository;
+
+    /** @var PasswordVerifier */
+    private $passwordVerifier;
+
+    public function __construct(
+        SessionInterface $session,
+        UserRepository $userRepository
+    ) {
+        $this->session = $session;
+        $this->userRepository = $userRepository;
+    }
+
+    public function loginUser(string $username, string $password): bool
+    {
+        $this->session->set("admin-user-id", null);
+        $user = $this->getUserByUsernameAndPassword($username, $password);
+        if ($user === null) {
+            return false;
+        }
+        $userId = $user->getId();
+        if ($userId === null) {
+            $userId = -1;
+        }
+        $this->session->set("admin-user-id", $userId);
+        return true;
+    }
+
+    private function logoutUser(): void
```

Comment:
Tato metoda by měla být také public. Ale pro tuto chvíli to nevadí, mergneme to i tak.